common/src/main/java/com/yapp/common/base/BaseViewModel.kt

```diff
@@ -41,12 +41,16 @@ abstract class BaseViewModel<S : UiState, A : UiSideEffect, E : UiEvent>(
         _uiState.value = state
     }
 
-    protected fun setEffect(builder: () -> A) {
-        val effectValue = builder()
-        viewModelScope.launch { _effect.send(effectValue) }
+    open fun setEffect(vararg builder: A) {
```

Comment:
```suggestion
    protected fun setEffect(vararg builder: A) {
```
요렇게 해서 ui에서는 접근 못하도록 하는게 좋을것 같아요!

Reply:
2af0dc6389127404d740570970910bfe4e9f63e4